manual/src/refman/modtypes.etex

```diff
@@ -280,14 +280,17 @@ one may use the abbreviated form
 Assuming @module-type@ denotes a signature, the expression
 @module-type 'with' mod-constraint@ @{ 'and' mod-constraint }@ denotes
 the same signature where type equations have been added to some of the
-type specifications, as described by the constraints following the
-"with" keyword. The constraint @'type' [type-parameters] typeconstr
+type specifications and some new components have been added to submodules,
+as described by the constraints following the "with" keyword.
+The constraint @'type' [type-parameters] typeconstr
 '=' typexpr@  adds the type equation @'=' typexpr@ to the specification
 of the type component named @typeconstr@ of the constrained signature.
 The constraint @'module' module-path '=' extended-module-path@ adds
 type equations to all type components of the sub-structure denoted by
 @module-path@, making them equivalent to the corresponding type
-components of the structure denoted by @extended-module-path@.
```

Comment:
I think it'd be easier to describe this operation as turning `S with module M = N` (the example a few lines down) into `sig type t module M : module type of N end` (provided the correct subtype relationship holds, which is sort of stated below).

Because not only can type equations be added to existing definitions, and module components be added, but `module A : Signature` items by replaced by `module A = Alias` items (and I suspect you could have `val` -> `external` replacements as well, but didn't check). The `module type of` description handles this, whereas the current text doesn't.

Now, sadly, `module type of` hasn't been defined at this point in the manual. So maybe we can paraphrase instead ("the inferred type of N")?

Reply:
I think we can use `inferred module type of` in the text (with a link to `module type of`?):
```
The constraint @'module' module-path '=' extended-module-path@ refines the module type of
@module-path@` to the inferred module type\footnote{see section~\ref{s:module-type-of}}
of the module @extended-module-path@.
```

Reply:
I like the phrasing. I don't think we should point to the definition of `module type of` unless we use `module type of` as a module expression. The [manual page](https://v2.ocaml.org/manual/moduletypeof.html) just says it expands to the module type inferred for the module expression, which we already say.

(well and in fact, the `module type of` page also says `To make this module type reusable in many situations, it is intentionally not strengthened`, so that's been wrong for several releases)

I think the earlier sentence you modified could be tweaked similarly.